instill/model/v1alpha/model.proto

```diff
@@ -0,0 +1,372 @@
+  }
+
+  // Status is the instance of the enum type ServingStatus
+  ServingStatus status = 1;
+}
+
+// ReadinessRequest represents a request to check a service readiness status
+message ReadinessRequest {
+  // Service name to check for its readiness status
+  string service = 1;
+}
+
+// ReadinessResponse represents a response for a service readiness status
+message ReadinessResponse {
+  // ServingStatus enumerates the status of a queried service
+  enum ServingStatus {
+    // Serving status: UNSPECIFIED
+    SERVING_STATUS_UNSPECIFIED = 0;
+    // Serving status: SERVING
+    SERVING_STATUS_SERVING = 1;
+    // Serving status: NOT SERVING
+    SERVING_STATUS_NOT_SERVING = 2;
+  }
+
+  // Status is the value of ServingStatus enum type
+  ServingStatus status = 1;
+}
+
+// ModelVersion represents the version of a model
+message ModelVersion {
+
+  // Status enumerates the status of a model version
+  enum Status {
+    // Status: UNSPECIFIED
+    STATUS_UNSPECIFIED = 0;
+    // Status: OFFLINE
+    STATUS_OFFLINE = 1;
+    // Status: ONLINE
+    STATUS_ONLINE = 2;
+    // Status: ERROR
+    STATUS_ERROR = 3;
+  }
+
+  // Model version
+  uint64 version = 1;
+  // Model version corresponding model ID
+  uint64 model_id = 2;
+  // Model version description
+  string description = 3;
+  // Model version creation time
+  google.protobuf.Timestamp created_at = 4;
+  // Model version update time
+  google.protobuf.Timestamp updated_at = 5;
+  // Model version status
+  Status status = 6;
+}
+
+// Model represents the content of a model
+message Model {
+
+  // Task enumerates the task type of a model
+  enum Task {
+    // Task: UNSPECIFIED
+    TASK_UNSPECIFIED = 0;
+    // Task: CLASSIFICATION
+    TASK_CLASSIFICATION = 1;
+    // Task: DETECTION
+    TASK_DETECTION = 2;
+  }
+
+  // Model ID
+  uint64 id = 1 [ (google.api.field_behavior) = OPTIONAL ];
+  // Model name
+  string name = 2;
+  // Model full name (i.e., [user_name/org_name]/model_name)
+  string full_name = 3 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model task
+  Task task = 4;
+  // Model versions
+  repeated ModelVersion model_versions = 5;
+}
+
+// CreateModelRequest represents a request to create a model
+message CreateModelRequest {
+  // Model name
+  string name = 1 [ (google.api.field_behavior) = REQUIRED ];
+  // Model content in byte
+  bytes byte = 2 [ (google.api.field_behavior) = REQUIRED ];
+  // Model description
+  string description = 3 [ (google.api.field_behavior) = OPTIONAL ];
+  // Model task type
+  Model.Task task = 4 [ (google.api.field_behavior) = OPTIONAL ];
+}
+
+// CreateModelResponse represents a response for a model instance
+message CreateModelResponse {
+  // A model instance
+  Model model = 1 [ (google.api.field_behavior) = REQUIRED ];
+}
+
+// CreateModelRequest represents a request to update a model
+message UpdateModelRequest {
+  // A model instance
+  Model model = 1 [ (google.api.field_behavior) = REQUIRED ];
```

Comment:
@pinglin @xiaofei-du this RPC used for update model version then input need the updating model version info instead of model. The model here should be model name. What do you think?

Reply:
BTW, should we separate updated model version info into a separated message. Reason: some fields such as created_at, updated_at, etc is not allowed to be updated

Reply:
Right. That makes sense. I will resume back to the previous message content.